lambda/css-api/src/services/keycloak-service.ts

```diff
@@ -0,0 +1,422 @@
+import axios, { AxiosError, AxiosInstance } from 'axios';
+import { getKeycloakCredentials } from '../util';
+import jwt from 'jsonwebtoken';
+import { RolePayload } from '../types';
+import createHttpError from 'http-errors';
+import { inject, singleton } from 'tsyringe';
+
+interface KeycloakTokenResponse {
+  access_token: string;
+  refresh_token: string;
+  expires_in: number;
+  refresh_expires_in: number;
+  token_type: string;
+  'not-before-policy': number;
+  session_state: string;
+  scope: string;
+}
+
+export class KeycloakService {
+  private realm: string = 'standard';
+  private httpClient: AxiosInstance;
+  private environment: string;
+  private accessToken: string | null = null;
+  private refreshToken: string | null = null;
+  private refreshing: boolean = false;
+
+  constructor() {}
+
+  setEnvironment(environment: string) {
```

Comment:
Should this be in the constructor instead? I think this axios setup will rerun everytime KeycloakServiceFactory.getKeycloakService is called

Reply:
I am instantiating three Keycloak services (by each env) anyway so doesn't make a difference if I use constructor. We wouldn't require to re-instantiate keycloak anywhere in the code but request the instance from keycloak service factory, which is a singleton.